scipy/stats/tests/test_stats.py

```diff
@@ -3998,5 +3998,48 @@ def test_weighted_stouffer2(self):
                                      weights=np.array((1, 4, 9)))
         assert_approx_equal(p, 0.1464, significant=4)
 
+def test_jensen_shannon_divergence():
+    for _ in range(8):
```

Comment:
I think we normally try to fix the random seeds in the test suite. It may also be more paradigmatic to rewrite this with a pytest parametrization or fixture approach for the different "scenarios," though perhaps not crucial.